### ISSUE TITLE
Nesting 'with' fails

### ISSUE THREAD

**Issue by clintonb** [capistrano/sshkit#43]
I have some example code below from my Capistrano v3 tasks for Django. It seems that nesting 'with' calls is broken. Is this functionality supposed to be supported? The docs state "One will notice that it's quite low level, but exposes a convenient API, the as()/within()/with() are nestable in any order, repeatable, and stackable."

My current work around is to wrap the body of with_virtualenv in a begin-rescue block that captures the NameError and check for the message "instance variable @_env not defined".

Note that my intention is to keep the Django and virtualenv tasks in separate files/gems, hence my not simply merging the with calls.

```
def with_virtualenv(&block)
  with path: "#{fetch(:virtualenv_dir)}/bin:$PATH" do
    block.call
  end
end

def django_manage(*args)
  within release_path do
    with_virtualenv do
      with app_environment: fetch(:stage) do
        execute :python, 'manage.py', args
      end
    end
  end
end

# Results in "instance variable @_env not defined"
namespace :django do
  task :migrate do
    on primary fetch(:migration_role) do
      django_manage 'syncdb', '--noinput', '--migrate'
    end
  end
end
```

**Comment by DylanFrese** [comment]
Should be fixed with PR #345

**Comment by leehambley** [comment]
If someone can confirm, I'd be glad to close this. I'll swing by again in a few days and check.

**Comment by ArturT** [comment]
@leehambley  I had a problem during deploy with capistrano and whenever:update_crontab.

```
cap aborted!
NameError: instance variable @_env not defined

Tasks: TOP => whenever:update_crontab
```

I switched to sshkit master branch and the PR https://github.com/capistrano/sshkit/pull/345 that was merged solved my problem.

**Comment by ansonhoyt** [comment]
@leehambley I echo @ArturT, updating to master fixed it for me.

If it helps confirm, here's output with `--trace` before I switched to master:
```
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-linux]
** Invoke bundler:map_bins (first_time)
** Invoke passenger:bundler:hook (first_time)
** Execute passenger:bundler:hook
** Execute bundler:map_bins
** Invoke deploy:set_rails_env (first_time)
** Execute deploy:set_rails_env
** Invoke deploy:set_linked_dirs (first_time)
** Execute deploy:set_linked_dirs
** Invoke deploy:set_rails_env
** Invoke whenever:update_crontab (first_time)
** Execute whenever:update_crontab
00:00 whenever:update_crontab
      01 /usr/local/rvm/bin/rvm ruby-2.3.0@pbis do bundle exec whenever --update-crontab papbs --set environment=demo --roles=web,app,db
      01 [write] crontab file updated
    ✔ 01 ahoyt@demo 3.447s
cap aborted!
NameError: instance variable @_env not defined
/Users/ahoyt/.rvm/gems/ruby-2.3.0@pbis/gems/sshkit-1.10.0/lib/sshkit/backends/abstract.rb:91:in `remove_instance_variable'
/Users/ahoyt/.rvm/gems/ruby-2.3.0@pbis/gems/sshkit-1.10.0/lib/sshkit/backends/abstract.rb:91:in `with'
/Users/ahoyt/.rvm/gems/ruby-2.3.0@pbis/gems/whenever-0.9.6/lib/whenever/capistrano/v3/tasks/whenever.rake:8:in `block (2 levels) in setup_whenever_task'
/Users/ahoyt/.rvm/gems/ruby-2.3.0@pbis/gems/sshkit-1.10.0/lib/sshkit/backends/abstract.rb:80:in `within'
/Users/ahoyt/.rvm/gems/ruby-2.3.0@pbis/gems/whenever-0.9.6/lib/whenever/capistrano/v3/tasks/whenever.rake:7:in `block in setup_whenever_task'
/Users/ahoyt/.rvm/gems/ruby-2.3.0@pbis/gems/sshkit-1.10.0/lib/sshkit/backends/abstract.rb:29:in `instance_exec'
/Users/ahoyt/.rvm/gems/ruby-2.3.0@pbis/gems/sshkit-1.10.0/lib/sshkit/backends/abstract.rb:29:in `run'
/Users/ahoyt/.rvm/gems/ruby-2.3.0@pbis/gems/sshkit-1.10.0/lib/sshkit/runners/parallel.rb:12:in `block (2 levels) in execute'
Tasks: TOP => whenever:update_crontab```

**Comment by mattbrictson** [comment]
Thanks for verifying! I'll close this issue since it has been fixed by PR #345.

I'll release 1.11.0 in a few hours.

**Comment by mattbrictson** [comment]
🚢 1.11.0
